harvester/src/run_daily_weather.py

```diff
@@ -0,0 +1,158 @@
+import sys
+import psycopg2.extras
+import psycopg2
+from dotenv import load_dotenv
+import logging
+import os
+import requests
+import datetime
+
+# This script fetches hourly weather data from the BrightSky API, aggregates it to daily weather data and stores it in the database
+
+# Set up logging
+logging.basicConfig()
+logging.root.setLevel(logging.INFO)
+
+# Load the environmental variables
+load_dotenv()
+
+# Check if all required environmental variables are accessible
+for env_var in [
+    "PG_SERVER",
+    "PG_PORT",
+    "PG_USER",
+    "PG_PASS",
+    "PG_DB",
+    "WEATHER_HARVEST_LAT",
+    "WEATHER_HARVEST_LNG",
+]:
+    if env_var not in os.environ:
+        logging.error("❌Environmental Variable {} does not exist".format(env_var))
+        sys.exit(1)
+
+PG_SERVER = os.getenv("PG_SERVER")
+PG_PORT = os.getenv("PG_PORT")
+PG_USER = os.getenv("PG_USER")
+PG_PASS = os.getenv("PG_PASS")
+PG_DB = os.getenv("PG_DB")
+WEATHER_HARVEST_LAT = os.getenv("WEATHER_HARVEST_LAT")
+WEATHER_HARVEST_LNG = os.getenv("WEATHER_HARVEST_LNG")
+
+# Establish database connection
+try:
+    database_connection_str = f"host='{PG_SERVER}' port={PG_PORT} user='{PG_USER}' password='{PG_PASS}' dbname='{PG_DB}'"
```

Comment:
Just nitpicking _but:_ Why use this connection string that holds the env vars? IMHO this is more error prone than using the named arguments in the method See https://www.psycopg.org/docs/module.html#psycopg2.connect

Reply:
changed to use named arguments